### PR TITLE
Continue backup operations per default in case of errors

### LINKED ISSUE #54: continue backup operations in case of source failure
I have some files in my homedir which are not readable by my user (leftovers from some build process under a different user). When knoxite encounters those files it just stops. It would be great if operations could pursue anyway (logging the error).
I think this should be the default behaviour with a 'stop on error' option.

### PR DESCRIPTION
**Pull request by penguwin**
This pr brings support for ongoing backup operations in knoxite. In case of an error knoxite will now print out the error, but pursue the current backup operation. Eventually, it will print out a summary of all errors when the operation finishes.

You can still choose to immediately exit on the first error with the `--pedantic` command line flag or - per repo - in the configuration file.

This also should fix #54.

### CODE REVIEW

**Review by ma-hartma — APPROVED** [review]
Great contribution! Looks good to me.

File: cmd/knoxite/restore.go [hunk 1]
```diff
@@ -71,53 +77,62 @@ func init() {
...
-			if p.Error != nil {
+	errs := make(map[string]error)
+	for p := range progress {
+		if p.Error != nil {
+			if restoreOpts.Pedantic {
 				fmt.Println()
 				return p.Error
 			}
+			errs[p.Path] = p.Error
+			stats.Errors += 1
```
Comment (muesli) [inline cmd/knoxite/restore.go:106]: `stats.Errors++` would suffice.

File: cmd/knoxite/store.go [hunk 2]
```diff
@@ -86,6 +90,7 @@ func initStoreFlags(f func() *pflag.FlagSet, opts *StoreOptions) {
 	f().StringVarP(&opts.Encryption, "encryption", "e", "", "encryption algo to use: aes (default), none")
 	f().UintVarP(&opts.FailureTolerance, "tolerance", "t", 0, "failure tolerance against n backend failures")
 	f().StringArrayVarP(&opts.Excludes, "excludes", "x", []string{}, "list of excludes")
+	f().BoolVar(&opts.Pedantic, "pedantic", false, "continue on failue")
```
Comment (muesli) [inline cmd/knoxite/store.go:93]: Typo in `failue`.

File: cmd/knoxite/store.go [hunk 3]
```diff
@@ -152,8 +159,11 @@ func store(repository *knoxite.Repository, chunkIndex *knoxite.ChunkIndex, snaps
 
 		default:
 			if p.Error != nil {
-				fmt.Println()
-				return p.Error
+				if storeOpts.Pedantic {
+					return p.Error
+				}
+				errs[p.Path] = p.Error
+				snapshot.Stats.Errors += 1
```
Comment (muesli) [inline cmd/knoxite/store.go:166]: `snapshot.Stats.Errors++` would suffice.

File: cmd/knoxite/store.go [hunk 4]
```diff
@@ -152,8 +159,11 @@ func store(repository *knoxite.Repository, chunkIndex *knoxite.ChunkIndex, snaps
 
 		default:
 			if p.Error != nil {
-				fmt.Println()
-				return p.Error
+				if storeOpts.Pedantic {
+					return p.Error
```
Comment (muesli) [inline cmd/knoxite/store.go:163]: `fmt.Println()` removed accidentally.

**Review by muesli — CHANGES_REQUESTED** [review]
Almost there!

**Review by ma-hartma — APPROVED** [review]

**Review by muesli — APPROVED** [review]
Fantastic, I think we can merge this!

**Pull request merged**